app/code/Magento/Theme/Model/Menu/MenuNode.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Theme\Model\Menu;
+
+use Magento\Theme\Api\Data\MenuNodeInterface;
+
+/**
+ * Data container for menu item.
+ */
+class MenuNode implements MenuNodeInterface
```

Comment:
I would propose to extend "\Magento\Framework\DataObject" to be consistent with other similar classes in Magento 2 modules